spec/support/corpus.rb

```diff
@@ -226,6 +245,60 @@ def system(arguments)
         end
       end
 
+      # Mapping of files which we expect to cause errors during mutation generation
+      class ErrorWhitelist
+        class UnnecessaryExpectation < StandardError
+          MESSAGE = 'Expected to encounter %s while mutating "%s"'.freeze
+
+          def initialize(*error_info)
+            super(MESSAGE % error_info)
+          end
+        end
+
+        include Concord.new(:map), Adamantium
+
+        # Assert that we expect to encounter the provided exception for this path
+        #
+        # @param path [Pathname]
+        # @param exception [Exception]
+        #
+        # @raise provided exception if we are not expecting this error
+        #
+        # This method is reraising exceptions but rubocop can't tell
+        # rubocop:disable Style/SignalException
+        #
+        # @return [undefined]
+        def assert_error(path, exception)
+          original_error = exception.cause || exception
+
+          raise exception unless map.fetch(original_error.inspect, []).include?(path)
+        end
+
+        # Assert that we expect to not encounter an error for the specified path
+        #
+        # @param path [Pathname]
+        #
+        # @raise [UnnecessaryExpectation] if we are expecting an exception for this path
+        #
+        # @return [undefined]
+        def assert_success(path)
+          map.each do |error, paths|
+            next unless paths.include?(path)
```

Comment:
I tend to only use guard clauses if the code I am guarding is more than one line of code. In this case you could probably write this as a one-liner:

```ruby
map.each do |error, paths|
  fail UnnecessaryExpectation.new(error, path) if paths.include?(path)
end
```